src/traces/sankey/attributes.js

```diff
@@ -9,13 +9,16 @@
 'use strict';
 
 var fontAttrs = require('../../plots/font_attributes');
+var plotAttrs = require('../../plots/attributes');
 var colorAttrs = require('../../components/color/attributes');
 var fxAttrs = require('../../components/fx/attributes');
 var domainAttrs = require('../../plots/domain').attributes;
 
 var overrideAll = require('../../plot_api/edit_types').overrideAll;
 
-var attrs = module.exports = overrideAll({
+module.exports = overrideAll({
+    hoverinfo: plotAttrs.hoverinfo,
```

Comment:
We'll need to extend the default `hoverinfo` declaration 

https://github.com/plotly/plotly.js/blob/36d9fb4ad341510823e6e25cb9202867ac94a94a/src/plots/attributes.js#L117-L130

with new flags. I'm thinking `'none'`, `'skip'` and `'all'`(the default) which I think should resolve @alexcjohnson 's concern https://github.com/plotly/plotly.js/pull/3096#discussion_r227626370

We should probably have a special description too to explain how trace `hoverinfo` can propagate to the node/link `hoverinfo`.

Reply:
@etpinard  I'm having trouble setting `hoverinfo.flags` to an empty array `[ ]`. Running `Plotly.PlotSchema.get().traces.sankey.attributes.hoverinfo` always return 5 flags.

The following doesn't work:
```
var attrs = module.exports = overrideAll({
    hoverinfo: plotAttrs.hoverinfo,
    ...
})
attrs.hoverinfo.flags = []
```

The following also doesn't work:
```
var attrs = module.exports = overrideAll({
    hoverinfo: extendFlat({}, plotAttrs.hoverinfo, {
        flags: ['random']
    }),
    ...
})
```
although it replaces the first array element with `'random'`.

Reply:
see https://github.com/plotly/plotly.js/issues/3058#issuecomment-429069306 - it's just a problem with the schema, the modules themselves are fine.

Reply:
> the modules themselves are fine.

Right, `gd._fullData[0]._module.attributes.hoverinfo.flags` should have the new list.

Reply:
Thank you guys for the quick reply! But won't this be an issue for the automatically generated documentation?

Reply:
sure will 😅

Reply:
and already is - feel like tackling #3058, or at least that part of it?

Reply:
That or, remove `hoverinfo` from the global (plots/attributes.js) trace attributes.

Reply:
@alexcjohnson I will look into #3058! I actually wanted to improve on plot.ly/javascript/reference so I guess that would be a good start for me since they're related :)